yasa/plotting.py

```diff
@@ -54,79 +41,96 @@ def plot_hypnogram(hypno, sf_hypno=1 / 30, lw=1.5, fill_color=None, ax=None):
     --------
     .. plot::
 
-        >>> import yasa
-        >>> import numpy as np
+        >>> from yasa import Hypnogram
         >>> import matplotlib.pyplot as plt
-        >>> hypno = np.loadtxt("https://github.com/raphaelvallat/yasa/raw/master/notebooks/data_full_6hrs_100Hz_hypno_30s.txt")
-        >>> fig, ax = plt.subplots(1, 1, figsize=(7, 3), constrained_layout=True)
-        >>> ax = yasa.plot_hypnogram(hypno, fill_color="gainsboro", ax=ax)
+        >>> values = 4 * ["W", "N1", "N2", "N3", "REM"] + ["ART", "N2", "REM", "W", "UNS"]
+        >>> hyp = Hypnogram(values, freq="24min").upsample("30s")
+        >>> ax = hyp.plot_hypnogram(lw=2, highlight="REM", fill_color="thistle")
+        >>> plt.tight_layout()
+
+    .. plot::
+
+        >>> from yasa import simulate_hypno
+        >>> fig, axes = plt.subplots(nrows=2, figsize=(6, 4), constrained_layout=True)
+        >>> hyp_a = simulate_hypno(tib=90, n_stages=3, seed=99)
+        >>> hyp_b = simulate_hypno(tib=90, n_stages=3, seed=99, start="2022-01-31 23:30:00")
+        >>> hyp_a.plot_hypnogram(fill_color="whitesmoke", ax=axes[0])
+        >>> hyp_b.plot_hypnogram(fill_color="whitesmoke", ax=axes[1])
     """
+    from yasa import Hypnogram  # Avoiding circular import
+
+    assert isinstance(hyp, Hypnogram), "`hypno` must be YASA Hypnogram."
+
     # Increase font size while preserving original
     old_fontsize = plt.rcParams["font.size"]
     plt.rcParams.update({"font.size": 18})
 
-    # Safety checks
-    assert isinstance(hypno, (np.ndarray, pd.Series, list)), "hypno must be an array."
-    hypno = np.asarray(hypno).astype(int)
-    assert (hypno >= -2).all() and (hypno <= 4).all(), "hypno values must be between -2 to 4."
-    assert hypno.ndim == 1, "hypno must be a 1D array."
-    assert isinstance(sf_hypno, (int, float)), "sf must be int or float."
-
-    bins = np.arange(hypno.size + 1) / (sf_hypno * 3600)
-    # Make sure that REM is displayed after Wake
-    hypno = pd.Series(hypno).map({-2: -2, -1: -1, 0: 0, 1: 2, 2: 3, 3: 4, 4: 1}).values
-
-    # Reduce data and bin edges to only moments of change in the hypnogram
-    # (to avoid drawing thousands of tiny individual lines when sf is high)
-    change_points = np.nonzero(np.ediff1d(hypno, to_end=1))
-    hypno = hypno[change_points]
-    bins = np.append(0, bins[change_points])
-
-    # Make masks for REM and Artefact/Unscored
-    hypno_rem = np.ma.masked_not_equal(hypno, 1)
-    hypno_art_uns = np.ma.masked_greater(hypno, -1)
-    hypno_sleep = np.ma.masked_less(hypno, 0)
+    ## Remap stages to be in desired y-axis order ##
+    # Start with default of all allowed labels
+    stage_order = hyp.labels.copy()
+    stages_present = hyp.hypno.unique()
```

Comment:
Oh this is good to know. I knew pandas returned in order of presentation, but didn't realize that numpy did alphabetical.

I think it's fine to keep pandas, since actually it's easier if WAKE is first (ie, order of presentation). This is why I swapped out the order of `accepted` 2-stage values in `yasa.Hypnogram`. I figured that was otherwise arbitrary, so now WAKE is before SLEEP.